test/CMakeLists.txt

```diff
@@ -59,7 +59,7 @@ elseif( "${CMAKE_CXX_COMPILER_ID}" MATCHES "GNU" OR
         target_compile_options( gsl-lite-cpp14.t  PUBLIC -std=c++14 )
     endif()
 
-    add_compile_options( -Wall -Wno-missing-braces -fno-elide-constructors )
+	set ( CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wall -Wno-missing-braces -fno-elide-constructors -Wconversion -Wsign-conversion -Wno-string-conversion" )
```

Comment:
I looked it up, this is the issue with `add_compile_options`:

> Adds options to the compiler command line for targets in the current directory and below that are added **after** this command is invoked.

[From this stack overflow answer](http://stackoverflow.com/a/40522384)
Isn't CMake wonderfully intuitive?

So it seems the `CMakeLists.txt` needs to be a bit rearranged in general to fix this.